src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -288,6 +305,60 @@ internal void TakeSnapShot(bool outputDelta)
                         break;
                     }
             }
+
+            this.MetricPointStatus = MetricPointStatus.NoPendingCollect;
```

Comment:
This might be not thread-safe. If the snapshot(value read) was done, and before SnapShot updates the status to NoPendingCollect, another Update might have update the value,status, then this will overwrite the status, and we'll miss that update, as it wont be covered in next snapshot.

Reply:
For histogram - this is straightforward - move the `this.MetricPointStatus = MetricPointStatus.NoPendingCollect;` inside the lock we already take anyway for Update/Snapshot.

Reply:
For Gauge, I think the following might do the trick:
Update - same as now
SnapShot  - 
1. Read the value.
2. Update Status to NoPendingCollect
3. Check if value changed since step 1. (i.e some other update(s) occurred since step 1)
4. If yes, revert status to PendingCollect. (this will ensure that the updates we missed, will be caught in next cycle)
5. Else, move on.


I think the same can be used for SUM as well.